docs/proposals/scheduling/average-replicas-assign/average-replicas-assign.md

```diff
@@ -0,0 +1,263 @@
+
+<!--
+在 assignFuncMap 中，我们需要添加一个 AverageReplicas 分配策略，其能够借助 estimator 提供的能力感知目前各个工作集群还有多少资源可用，同时在此基础上尽力将副本平均分配给各个所选出的集群。
+-->
+
+The replica allocation strategy can be verified by checking the logs and the instance running status (i.e., which cluster the instances were allocated to and whether they are Running).
+
+<!--
+可以通过日志即实例运行情况（即实例被分配到了哪个集群以及是否 Running）来确认副本分配策略是否符合预期。
+-->
+
+### User Stories 
+
+#### Story 1 
+
+For example, as developers, we have a service with 6 replicas and three clusters. We hope Karmada can perform high-availability deployment across clusters. With the support of the AverageReplicas allocation strategy, we can ensure that each cluster has 2 replica instances, as shown in the figure below:
+
+<!--
+例如，作为开发人员，我们有一个包含 6 个副本的服务且有三个集群，希望 Karmada 能够进行跨集群的高可用性部署。那么在副本分配策略 AverageReplicas 的支持下，我们就能够保证一个集群上有 2 个副本实例，如下图所示：
+-->
+
+![user-story](user-story.png)
+
+### Risks and Mitigations
+
+The changes introduced in this proposal will add an AverageReplicas allocation strategy to Karmada. By default, this strategy will not be active, as users have not directly specified the use of the AverageReplicas strategy. If users want to use this feature, they must explicitly specify the AverageReplicas replica allocation strategy.
+
+<!--
+该 Proposal 所新增的改动会为 Karmada 添加一个 AverageReplicas 分配策略，默认情况下，该策略是不会生效的，因为用户并没有直接指定使用 AverageReplicas 策略。如果用户想要使用这个功能，则必须明确指定 AverageReplicas 副本分配策略。
+-->
+
+Therefore, the introduction of this change in Karmada is compatible with previous user configurations. This proposal does not present any foreseeable risks at this time.
+
+<!--
+因此，Karmada 引入这项改动是能够兼容过往用户的配置的，该 Proposal 目前不存在任何可预见性的风险。
+-->
+
+## Design Details
+
+### API Design
+
+In Karmada, the API related to scheduling is the PropagationPolicy. Currently, its API design is as follows:
+
+<!--
+在 Karmada 中，和调度相关的 API 即为 PropagationPolicy，目前，其 API 设计如下：
+-->
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy 
+metadata:
+spec:
+	placement:
+		replicaScheduling: 
+			replicaDivisionPreference: Average
+```
+
+As we can see, to add an AverageReplicas replica allocation strategy, we can consider adding a new value, "Average," to the replicaDivisionPreference.
+
+<!--
+可以看到，如果要添加一个 AverageReplicas 副本分配策略，可以考虑给 replicaDivisionPreference 添加一个新的取值 Average。
+-->
+
+Note that `replicaDivisionPreference == Average` will only be effective when `replicaSchedulingType == Divided`. Additionally, since `selectBestClustersByRegion` currently does not consider whether the selected clusters can accommodate the corresponding number of replica resources, **the AverageReplicas strategy is only applicable to `selectBestClustersByCluster`.**
```

Comment:
I don't get it, we should only focus on the `Assign` phase, as for what the spread constraints are, we don't care.